EmployeeManagement/Sources/openai_request.swift

```diff
@@ -0,0 +1,70 @@
+import Foundation
+
+// Hardcoded OpenAI API Key (not recommended)
+let apiKey = "sk-utRJzHluRSPOljnDhnORT5BlbkFJkPSHrIf6MqUIs1rAfUY0"
+
+// API Endpoint
+let url = URL(string: "https://api.openai.com/v1/completions")!
+
+// Data to send in the API request
+let requestData: [String: Any] = [
+    "model": "text-davinci-003",
+    "prompt": "Once upon a time",
+    "max_tokens": 50
+]
+
+// Serialize request data to JSON
+guard let jsonData = try? JSONSerialization.data(withJSONObject: requestData, options: []) else {
+    print("Error: Unable to serialize request data.")
+    exit(1)
+}
+
+// Prepare the URLRequest
+var request = URLRequest(url: url)
+request.httpMethod = "POST"
+request.setValue("Bearer \(apiKey)", forHTTPHeaderField: "Authorization")
+request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+request.httpBody = jsonData
+
+// Create a semaphore to wait for the async task
+let semaphore = DispatchSemaphore(value: 0)
+
+// Perform the request
+let task = URLSession.shared.dataTask(with: request) { data, response, error in
+    defer { semaphore.signal() }
+
+    if let error = error {
+        print("Error: \(error)")
+        return
+    }
+
+    guard let data = data else {
+        print("No data received")
+        return
+    }
+
+    // Decode and print the response
+    do {
+        if let json = try JSONSerialization.jsonObject(with: data, options: []) as? [String: Any],
+           let choices = json["choices"] as? [[String: Any]],
+           let text = choices.first?["text"] as? String {
+            print("Response from OpenAI:")
+            print(text)
+        } else {
+            print("Unexpected response format")
+            if let responseString = String(data: data, encoding: .utf8) {
+                print("Response: \(responseString)")
+            }
+        }
+    } catch {
+        print("Error decoding response: \(error)")
+        if let responseString = String(data: data, encoding: .utf8) {
+            print("Response: \(responseString)")
+        }
+    }
+}
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider extracting response handling logic</b></div>
<div id="fix">

The completion handler contains complex nested logic. Consider extracting the response handling into a separate function for better readability and maintainability.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -32,34 +32,38 @@
  let task = URLSession.shared.dataTask(with: request) { data, response, error in
      defer { semaphore.signal() }
 -    if let error = error {
 -        print("Error: \(error)")
 -        return
 -    }
 -    guard let data = data else {
 -        print("No data received")
 -        return
 -    }
 -    do {
 -        if let json = try JSONSerialization.jsonObject(with: data, options: []) as? [String: Any],
 -           let choices = json["choices"] as? [[String: Any]],
 -           let text = choices.first?["text"] as? String {
 -            print("Response from OpenAI:")
 -            print(text)
 -        } else {
 -            print("Unexpected response format")
 -            if let responseString = String(data: data, encoding: .utf8) {
 -                print("Response: \(responseString)")
 -            }
 -        }
 -    } catch {
 -        print("Error decoding response: \(error)")
 -        if let responseString = String(data: data, encoding: .utf8) {
 -            print("Response: \(responseString)")
 -        }
 -    }
 +    handleResponse(data: data, error: error)
  }
 +
 +private func handleResponse(data: Data?, error: Error?) {
 +    if let error = error {
 +        print("Error: \(error)")
 +        return
 +    }
 +    guard let data = data else {
 +        print("No data received")
 +        return
 +    }
 +    parseAndPrintResponse(data: data)
 +}

```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rishabhbohra3/Bito-CRA-Issues-Dataset/pull/10#issuecomment-2514547000>#0b17a0</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged